docs/sources/upgrade-guide/upgrade-v12.0/index.md

```diff
@@ -0,0 +1,79 @@
+---
+description: Guide for upgrading to Grafana v12.0
+keywords:
+  - grafana
+  - configuration
+  - documentation
+  - upgrade
+  - '12.0'
+title: Upgrade to Grafana v12.0
+menuTitle: Upgrade to v12.0
+weight: 600
+---
+
+# Upgrade to Grafana v12.0
+
+{{< docs/shared lookup="upgrade/intro.md" source="grafana" version="<GRAFANA_VERSION>" >}}
+
+{{< docs/shared lookup="back-up/back-up-grafana.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+1" >}}
+
+{{< docs/shared lookup="upgrade/upgrade-common-tasks.md" source="grafana" version="<GRAFANA_VERSION>" >}}
+
+## Technical notes
+
+### Grafana data source UID format enforcement
+
+**Ensure that your data source UIDs follow the correct standard**
+
+We've had standard ways to define UIDs for Grafana objects for years (at least [since Grafana v5](https://github.com/grafana/grafana/issues/7883)). While all of our internal code complies to this format, we haven't strictly enforced this format in REST APIs and provisioning paths that allow the creation and update of data sources.
+
+In Grafana v11.1, we [introduced](https://github.com/grafana/grafana/pull/86598) a warning that is sent to Grafana server logs every time a data source instance is created or updated using an invalid UID format.
+
+In Grafana v11.2, we [added](https://github.com/grafana/grafana/pull/89363/files) a new feature flag called `failWrongDSUID` that is turned off by default. When enabled, the REST APIs and provisioning reject any requests to create or update data source instances that have an incorrect UID.
+
+In Grafana v12.0, we're turning the feature flag `failWrongDSUID` on by default.
+
+#### Correct UID format
+
+You can find the exact regex definition [in the `grafana/grafana` repository](https://github.com/grafana/grafana/blob/c92f5169d1c83508beb777f71a93336179fe426e/pkg/util/shortid_generator.go#L32-L45).
+
+A data source UID can only contain:
+
+- Latin characters (`a-Z`)
+- Numbers (`0-9`)
+- Dash symbols (`-`)
+
+#### How do I know if I'm affected?
+
+- You can fetch all your data sources using the `/api/datasources` API. Review the `uid` fields, comparing them to the correct format, as shown [in the docs](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/#get-all-data-sources). The following script can help, but note that it's missing authentication that you [have to add yourself](https://grafana.com/docs/grafana/latest/developers/http_api/#authenticating-api-requests):
+
+```
+curl http://localhost:3000/api/datasources | jq '.[] | select((.uid | test("^[a-zA-Z0-9\\-_]+$") | not) or (.uid | length > 40)) | {id, uid, name, type}'
```

Comment:
Use `curl -s` to prevent output like this:

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7419    0  7419    0     0   809k      0 --:--:-- --:--:-- --:--:--  905k
```